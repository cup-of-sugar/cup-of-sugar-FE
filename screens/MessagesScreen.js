import * as WebBrowser from "expo-web-browser";
import * as React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Colors from "../constants/Colors";
import Inbox from "../components/Inbox";
import Outbox from "../components/Outbox";
import { ScrollView } from "react-native-gesture-handler";
import { gql } from "apollo-boost";
import { useQuery } from "@apollo/react-hooks";

export default function MessagesScreen(props) {
  const userId = props.route.params.userId;
  const [messageView, setMessageView] = React.useState("inbox");

  return (
    <View style={styles.container}>
      <ScrollView
        scrollIndicatorInsets={{ right: 1 }}
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
      >
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.inboxLabel}
            onPress={() => setMessageView("inbox")}
          >
            <Text style={styles.inboxMailboxText}>Inbox</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.sentLabel}
            onPress={() => setMessageView("sent")}
          >
            <Text style={styles.sentMailboxText}>Sent</Text>
          </TouchableOpacity>
        </View>
        {messageView === "inbox" ? (
          <View style={styles.inbox}>
            <Inbox userID={userId} navigation={props.navigation} />
          </View>
        ) : (
          <View style={styles.sent}>
            <Outbox userID={userId} navigation={props.navigation} />
          </View>
        )}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    paddingTop: 20,
    height: "100%"
  },
  contentContainer: {
    padding: 15
  },
  buttonContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between"
  },
  inboxLabel: {
    backgroundColor: Colors.lightBlue,
    borderTopLeftRadius: 7,
    borderTopRightRadius: 7,
    height: 70,
    padding: 20,
    width: 180
  },
  sentLabel: {
    backgroundColor: Colors.darkBlue,
    borderTopLeftRadius: 7,
    borderTopRightRadius: 7,
    height: 70,
    padding: 20,
    width: 180
  },
  inbox: {
    backgroundColor: Colors.lightBlue,
    borderBottomLeftRadius: 7,
    borderBottomRightRadius: 7,
    height: 500,
    paddingVertical: 20
  },
  sent: {
    backgroundColor: Colors.darkBlue,
    borderBottomLeftRadius: 7,
    borderBottomRightRadius: 7,
    height: 500,
    paddingVertical: 20
  },
  inboxMailboxText: {
    fontSize: 25,
    fontWeight: "bold",
    color: "#fff"
  },
  sentMailboxText: {
    fontSize: 25,
    fontWeight: "bold",
    color: "#fff",
    textAlign: "right"
  }
});
