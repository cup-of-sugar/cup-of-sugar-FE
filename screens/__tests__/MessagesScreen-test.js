import React from "react";
import MessagesScreen from "../MessagesScreen";
import { MockedProvider } from "@apollo/react-testing";
import ReactTestRenderer from "react-test-renderer";

it("should render without error if on borrowing path", () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: { params: { action: "borrow", userId: "4" } }
  };
  ReactTestRenderer.create(
    <MockedProvider mocks={[]}>
      <MessagesScreen {...props} />
    </MockedProvider>
  );
});

it("should render without error if on lending path", () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: { params: { action: "lend", userId: "4" } }
  };
  ReactTestRenderer.create(
    <MockedProvider mocks={[]}>
      <MessagesScreen {...props} />
    </MockedProvider>
  );
});
