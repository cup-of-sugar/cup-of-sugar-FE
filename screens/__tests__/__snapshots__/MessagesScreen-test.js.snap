// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "paddingTop": 50,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "padding": 15,
      }
    }
    scrollIndicatorInsets={
      Object {
        "right": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#0C94E1",
              "borderTopLeftRadius": 7,
              "borderTopRightRadius": 7,
              "height": 70,
              "opacity": 1,
              "padding": 20,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 25,
                "fontWeight": "bold",
              }
            }
          >
            Inbox (1)
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#385A94",
              "borderTopLeftRadius": 7,
              "borderTopRightRadius": 7,
              "height": 70,
              "opacity": 1,
              "padding": 20,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 25,
                "fontWeight": "bold",
                "textAlign": "right",
              }
            }
          >
            Sent (1)
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#0C94E1",
            "borderBottomLeftRadius": 7,
            "borderBottomRightRadius": 7,
            "height": 500,
            "paddingVertical": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderColor": "#fff",
              "borderRadius": 10,
              "borderWidth": 1,
              "flex": 0,
              "flexDirection": "row",
              "height": 70,
              "margin": 40,
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 20,
              "width": 350,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#385A94",
                "fontSize": 20,
                "fontWeight": "bold",
                "paddingRight": 10,
              }
            }
          >
            From: Tiger King
          </Text>
          <Image
            source={1}
            style={
              Object {
                "height": 25,
                "width": 115,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
